linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -2624,6 +2624,86 @@ public static boolean isMergeJoinSupported(JoinType joinType) {
     };
   }
 
+
+  /**
+   * A sort implementation optimized for a sort with a fetch size (LIMIT).
+   * @param offset how many rows are skipped from the sorted output.
+   *               Must be greater than or equal to 0.
+   * @param fetch how many rows are retrieved. Must be greater than 0.
+   */
+  public static <TSource, TKey> Enumerable<TSource> orderBy(
```

Comment:
Does it make sense to name this function as `topNOrderBy` or `topNSortLimit`?

Reply:
Might make sense, I'm not yet familiar how the Calcite project chooses names.

Reply:
Since the limit parameter is "optional" (it can take `Integer.MAX_VALUE` if `fetch` is null in `EnumerableLimitSort`, I think the current method name is ok.

Reply:
I see. That makes sense.